### PR TITLE
Rename from PerformanceTest to Wallet-gasCost, merge setupInitArgs with getInitData

### PR DESCRIPTION
**Pull request by JustynaBroniszewska**
...add deployProxy and remove duplicates

### CODE REVIEW

File: universal-login-contracts/lib/utils.ts [hunk 1]
```diff
@@ -6,6 +6,8 @@ export const createKey = () => {
   return {publicKey: address, privateKey};
 };
 
+export type Key = ReturnType<typeof createKey>;
```
Comment (marekkirejczyk) [inline universal-login-contracts/lib/utils.ts]: `KeyPair`

Comment (marekkirejczyk) [reply universal-login-contracts/lib/utils.ts]: Move to commons?

File: universal-login-contracts/lib/utils.ts [hunk 2]
```diff
@@ -6,6 +6,8 @@ export const createKey = () => {
...
+
```
Comment (marekkirejczyk) [inline universal-login-contracts/lib/utils.ts]: Rename file `encode.ts`.

File: universal-login-contracts/test/contracts/Wallet-gasCost.ts [hunk 3]
```diff
@@ -42,11 +40,10 @@ describe('Performance test', async () => {
   });
 
   it('Proxy deploy with ENS', async () => {
-    const initializeArgs = setupInitializeArgs({key: key.publicKey, ensAddress, resolverAddress, registrarAddress});
-    const initData = getInitWithENSData(initializeArgs);
+    const args = getDeployWithEnsArgs(key.publicKey, ensDomainData, walletMaster.address);
```
Comment (marekkirejczyk) [inline universal-login-contracts/test/contracts/Wallet-gasCost.ts]: `getProxyDeployWithENSArgs`

File: universal-login-contracts/test/fixtures/walletContract.ts [hunk 4]
```diff
@@ -4,54 +4,71 @@ import WalletMaster from '../../build/WalletMaster.json';
...
+type SetupInitializeArgs = {
+  key: string;
+  ensDomainData: EnsDomainData;
   name?: string;
   domain?: string;
-}
+};
 
-export function setupInitializeArgs({key, ensAddress, registrarAddress, resolverAddress, name = 'name', domain = 'mylogin.eth'}: SetupInitializeArgs) {
+export function setupInitializeArgs({key, ensDomainData, name = 'name', domain = 'mylogin.eth'}: SetupInitializeArgs) {
```
Comment (marekkirejczyk) [inline universal-login-contracts/test/fixtures/walletContract.ts]: `createInitializeWithENSArgs`

File: universal-login-contracts/lib/utils.ts [hunk 5]
```diff
@@ -6,6 +6,8 @@ export const createKey = () => {
...
 export const getInitWithENSData = (args: any[]) => new utils.Interface(WalletMaster.interface).functions.initializeWithENS.encode(args);
```
Comment (marekkirejczyk) [inline universal-login-contracts/lib/utils.ts]:
```
encodeInitalizeWithENSData(args : string[]) {
}
````

File: universal-login-relayer/test/helpers/createWalletContract.js [hunk 6]
```diff
@@ -12,7 +13,7 @@ export default async function createWalletContract(wallet, ensService) {
     wallet,
   );
   const walletArgs = [wallet.address, ...ensService.argsFor('alex.mylogin.eth')];
-  const initData = new utils.Interface(WalletMaster.interface).functions.initializeWithENS.encode(walletArgs);
+  const initData = getInitWithENSData(walletArgs);
```
Comment (marekkirejczyk) [inline universal-login-relayer/test/helpers/createWalletContract.js]: `encode...`

File: universal-login-contracts/lib/utils.ts [hunk 7]
```diff
@@ -6,6 +6,8 @@ export const createKey = () => {
...
 
 export const getInitData = (publicKey: string) => new utils.Interface(WalletMaster.interface).functions.initialize.encode([publicKey]);
```
Comment (marekkirejczyk) [inline universal-login-contracts/lib/utils.ts]: encodeInitalizeData(args : string[]) {

File: universal-login-contracts/lib/utils.ts (hunk 1 above)
Comment (JustynaBroniszewska) [reply universal-login-contracts/lib/utils.ts]: ✅

File: universal-login-contracts/lib/utils.ts (hunk 2 above)
Comment (JustynaBroniszewska) [reply universal-login-contracts/lib/utils.ts]: ✅

File: universal-login-contracts/test/contracts/Wallet-gasCost.ts (hunk 3 above)
Comment (JustynaBroniszewska) [reply universal-login-contracts/test/contracts/Wallet-gasCost.ts]: ✅

File: universal-login-contracts/test/fixtures/walletContract.ts (hunk 4 above)
Comment (JustynaBroniszewska) [reply universal-login-contracts/test/fixtures/walletContract.ts]: ✅

File: universal-login-contracts/lib/utils.ts (hunk 5 above)
Comment (JustynaBroniszewska) [reply universal-login-contracts/lib/utils.ts]: ✅

File: universal-login-relayer/test/helpers/createWalletContract.js (hunk 6 above)
Comment (JustynaBroniszewska) [reply universal-login-relayer/test/helpers/createWalletContract.js]: ✅

File: universal-login-contracts/lib/utils.ts (hunk 7 above)
Comment (JustynaBroniszewska) [reply universal-login-contracts/lib/utils.ts]: ✅

File: universal-login-contracts/test/fixtures/walletContract.ts [hunk 8]
```diff
@@ -2,56 +2,73 @@ import {providers, Wallet, utils} from 'ethers';
...
   const ensName = `${name}.${domain}`;
   const hashLabel = utils.keccak256(utils.toUtf8Bytes(name));
   const node = utils.namehash(ensName);
-  const args = [key, hashLabel, ensName, node, ensAddress, registrarAddress, resolverAddress];
+  const args = [key, hashLabel, ensName, node, ensDomainData.ensAddress, ensDomainData.registrarAddress, ensDomainData.resolverAddress];
   return args;
 }
 
+export function createProxyDeployWithENSArgs(publicKey: string, ensDomainData: EnsDomainData, walletMasterAddress: string) {
```
Comment (marekkirejczyk) [inline universal-login-contracts/test/fixtures/walletContract.ts]: Even better: `createArgsForProxyDeployWithENS`

File: universal-login-contracts/test/fixtures/walletContract.ts (hunk 4 above)
Comment (marekkirejczyk) [reply universal-login-contracts/test/fixtures/walletContract.ts]: even better: `createArgsForIntializeWithENS`

File: universal-login-contracts/test/fixtures/walletContract.ts [hunk 9]
```diff
@@ -2,56 +2,73 @@ import {providers, Wallet, utils} from 'ethers';
...
 
-export async function setupWalletContract(deployer: Wallet, contractOwner: Wallet) {
-  const {ensAddress, registrarAddress, resolverAddress, walletMaster, provider} = await setupEnsAndMaster(deployer);
-  const args = setupInitializeArgs({key: contractOwner.address, ensAddress, registrarAddress, resolverAddress});
-  const initData = getInitWithENSData(args);
-  const walletContract = await deployContract(deployer, ProxyContract, [walletMaster.address, initData]);
+async function deployProxy(deployer: Wallet, walletMasterAddress: string, ensDomainData: EnsDomainData, publicKey: string) {
+  const deployArgs = createProxyDeployWithENSArgs(publicKey, ensDomainData, walletMasterAddress);
+  const proxyContract = await deployContract(deployer, ProxyContract, deployArgs);
+  return proxyContract;
```
Comment (marekkirejczyk) [inline universal-login-contracts/test/fixtures/walletContract.ts]: Shorter:
```ts
return deployContract(deployer, ProxyContract, deployArgs)
```

File: universal-login-contracts/test/fixtures/walletMasterAndProxy.ts [hunk 10]
```diff
@@ -4,13 +4,14 @@ import MockToken from '../../build/MockToken.json';
 import MockContract from '../../build/MockContract.json';
 import {deployContract} from 'ethereum-waffle';
 import {utils, Contract, Wallet, providers} from 'ethers';
+import {encodeInitializeData} from '../../lib';
 
 const {parseEther} = utils;
 
 export default async function walletMasterAndProxy(unusedProvider : providers.Provider, [, , , , , , , , , wallet] : Wallet []) {
   const ownerWallet = Wallet.createRandom();
```
Comment (marekkirejczyk) [inline universal-login-contracts/test/fixtures/walletMasterAndProxy.ts]:
```ts
const keyPair = createKey();
```

File: universal-login-relayer/lib/services/WalletService.ts [hunk 11]
```diff
@@ -19,8 +19,8 @@ class WalletService {
   async create(key: string, ensName: string, overrideOptions = {}) {
     const ensArgs = this.ensService.argsFor(ensName);
     if (ensArgs !== null) {
-      let args = [key, ...ensArgs];
-      const initData = new utils.Interface(WalletMasterContract.interface).functions.initializeWithENS.encode(args);
+      let args = [key, ...ensArgs] as string[];
```
Comment (marekkirejczyk) [inline universal-login-relayer/lib/services/WalletService.ts]: In `ensService.ts`:
`argsFor(ensName: string) {` -> `argsFor(ensName: string) : string[] {`

**Pull request merged**